components/grid/state.md

```diff
@@ -10,923 +10,419 @@ position: 50
+| `Page` | `int?` | The current [page index]({%slug components/grid/features/paging%}). Some user actions reset the page index to 1, such as filtering or changing the page size. |
+| `SearchFilter` | `IFilterDescriptor` | The `CompositeFilterDescriptor` that holds the filter descriptors for the [`GridSearchBox`]({%slug grid-searchbox%}). |
+| `SelectedItems` | `ICollection<TItem>` | The currently [selected data item(s)]({%slug components/grid/selection/overview%}). |
+| `Skip` | `int?` | The number of scrolled data items when using [virtual row scrolling]({%slug components/grid/virtual-scrolling%}). In other words, this is the number of rows above the currently visible ones. |
+| `SortDescriptors` | `ICollection<SortDescriptor>` | The currently applied [sorts]({%slug components/grid/features/sorting%}). |
+| `TableWidth` | `string` | The sum of all visible column widths. This property changes together with `ColumnStates`. The `OnStateChanged` event does not fire separately for it. |
 
-* **SearchFilter** - filter descriptor specific to the GridSearchBox.
+\* `TItem` is the Grid model type.
 
-* **Grouping** - group descriptors (fields by which the grid is grouped), collapsed group indexes.
 
-* **Paging** - page index, offset (skip) for virtual scrolling.
+## Events
 
-* **Rows** - list of expanded items.
+The Grid features two events, which are related to its state.
 
-* **Sorting** - sort descriptors (fields by which the grid is sorted, and the direction).
+* [OnStateInit](#onstateinit)
+* [OnStateChanged](#onstatechanged)
 
-* **Selection** - list of selected items.
+### OnStateInit
 
-* **Columns** - Visible, Width, Index (order) of the column that the user sees, Locked (pinned).
+The `OnStateInit` event fires when the Grid is initializing. Use this event to:
 
-    * The grid matches the columns from its markup sequentially (in the same order) with the columns list in the state object. So, when you restore/set the state, the grid must initialize with the same collection of columns that were used to save the state.
-    
-        The `Index` field in the column state object represents its place (order) that the user sees and can choose through the `Reordable` feature, not its place in the grid markup. You can find an example below.
-    
-        If you want to change the visibility of columns, we recommend you use their `Visible` parameter rather than conditional markup - this parameter will be present in the state and will not change the columns collection count which makes it easier to reconcile changes.
-
-
-## Examples
+* Define initial state, for example default initial sorting;
+* Load and apply state that was previously saved in a database or in `localStorage`.
 
-You can find the following examples in this section:
+The generic event argument is of type `GridStateEventArgs<TItem>`. It has one important property and that is `GridState`. See [Information in the Grid State](#information-in-the-grid-state) for details about its members.
```

Comment:
"It has one important property" - this kind leads to the thought that there is only one property while the argument also includes `PropertyName`. It may be useful mentioning its purpose and stating it is always `null` when the Grid initializes.